src/c_tun_device.cpp

```diff
@@ -546,7 +548,8 @@ std::array<uint8_t, 6> c_tun_device_windows::get_mac(HANDLE handle) {
 
 void c_tun_device_windows::handle_read(const boost::system::error_code& error, std::size_t length) {
 	try {
-		if (error || (length < 1)) throw std::runtime_error(error.message());
+		if (length < 1) throw std::runtime_error("Readed 0 bytes from tun");
+		if (error) throw std::runtime_error(error.message());
 		if (length < 54) throw std::runtime_error("tun data length < 54"); // 54 == sum of header sizes
```

Comment:
ditto
This changes should be in separate PR better. (or at least change title of it)